packages/solana/src/client.ts

```diff
@@ -227,6 +229,17 @@ export class Web3Modal extends Web3ModalScaffold {
       }
     })
 
+    EventsController.subscribe(state => {
+      if (state.data.event === 'SELECT_WALLET' && state.data.properties?.name === 'Phantom') {
+        const isMobile = CoreHelperUtil.isMobile()
+        if (isMobile && !window.phantom) {
+          const href = window.location.href
+          const ref = `${href.startsWith('https') ? 'https' : 'http'}://${href.split('/')[2]}`
```

Comment:
For better readability, could we separate the parts and stay away from one-liner template literals with many conditions inside? This will make it more readable. 

```suggestion
      const href = window.location.href;
      const protocol = href.startsWith('https') ? 'https' : 'http';
      const host = href.split('/')[2];
      const ref = `${protocol}://${host}`;

      window.location.href = `https://phantom.app/ul/browse/${href}?ref=${ref}`;
```